src/components/cards/dialogs/allowedSps/AllowedSps.tsx

```diff
@@ -83,14 +85,40 @@ export const AllowedSPs: React.FC<ComponentProps> = ({
     })
   }
 
+  const isValidInput = (input: string): boolean => {
+    const regex = /^(f0)?\d+$/
+    return regex.test(input)
+  }
+
   const handleInputChange = (index: number, value: string): void => {
     const newData = [...data]
+    const updatedErrors = [...errors]
     newData[index] = value
+    value = value.trim()
+    clearTimeout(debounceTimer)
+    const timer = window.setTimeout(() => {
+      value = value.trim()
+      console.log(value)
+      if (!value) {
+        updatedErrors[index] = 'Value cannot be empty'
+        setErrors([...updatedErrors])
+        return
+      } else if (!isValidInput(value)) {
+        updatedErrors[index] = 'Invalid storage provider id'
+        setErrors([...updatedErrors])
```

Comment:
@tplocic20 what do you think about DYI form validation and debouncing? Is this the best way given the state of the app or is there some good lightweight lib we could include?

Reply:
Ideally we would use something like hookForm - but it's too much of a refactor I'm afraid